core/templates/dev/head/pages/community-dashboard-page/services/translate-text.service.spec.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Tests that the translate-text service is working as
+ * expected.
+ */
+
+require(
+  'pages/community-dashboard-page/services/' +
+  'translate-text.service.ts');
+
+describe('TranslateTextService', function() {
+  var TranslateTextService, $httpBackend;
+  var mockTranslateBackendObject;
+  var callbackObject;
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.inject(function($injector) {
+    $httpBackend = $injector.get('$httpBackend');
+    TranslateTextService = $injector.get('TranslateTextService');
+    mockTranslateBackendObject = {
+      version: 1,
+      state_names_to_content_id_mapping: {
+        Intro: {
+          content: 'IntroContent'
+        },
+        END: {
+          content: '<p>Congratulations, you have finished!</p>'
+        },
+        'Second Question': {
+          content: 'testContent'
+        }
+      }
+    };
+  }));
+
+  afterEach(function() {
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
+  describe('getTextToTranslate', function() {
+    it('should return text available for translation', function() {
+      $httpBackend.whenRoute('GET', '/gettranslatabletexthandler')
+        .respond(function(method, url, data, headers, params) {
+          return [200, mockTranslateBackendObject];
+        });
+      TranslateTextService.init(1, 'en', function() {
+        var translateData = TranslateTextService.getTextToTranslate();
+        expect(translateData.text).toEqual('IntroContent');
+        expect(translateData.more).toEqual(true);
+      });
+      $httpBackend.flush();
+    });
+  });
+
+  describe('suggestTranslatedText', function() {
+    it('should post new translated text data to the API', function() {
+      callbackObject = {
+        callback: function(data) {}
+      };
+      spyOn(callbackObject, 'callback');
+      $httpBackend.whenRoute('GET', '/gettranslatabletexthandler')
+        .respond(function(method, url, data, headers, params) {
+          return [200, mockTranslateBackendObject];
+        });
+      TranslateTextService.init(1, 'en', function() {
```

Comment:
I'm not sure what's going on here, I would suggest initializing the service for `describe('suggestTranslatedText'` (And for the above one as well) in beforeEach section and write a separate test for `initi` function.

Reply:
I tried doing that but it didn't work. There was some problem related to scope. And the way its written in the test seems to be the way it's supposed to be used in the code.
init is itself a function that the service exports. What's initi? Is that a typo?

Reply:
Generally `init` is for `initialization` i.e, we call this method to initialize the service with all the required data and once the service is initialized we can use it. (In this case, the text-service needs to be initialized based on the explorationId, so that it can fetch required text to translate.)

>  There was some problem related to scope.

Can you share the logs? maybe I can help.